lib/BackgroundJob/Tasks/CheckRequirementsTask.php

```diff
@@ -105,8 +97,37 @@ public function execute(FaceRecognitionContext $context) {
 				"If you already set this to unlimited, it seems your system is not having enough RAM memory.";
 			$this->logInfo($error_message);
 			return false;
+		} else {
+			$this->logDebug(sprintf('Found %d MB available to PHP.', $memory / (1024 * 1024)));
+			// No point going above 4GB ("4GB should be enough for everyone")
+			if ($memory > 4 * 1024 * 1024 * 1024) {
+				$this->logDebug('Capping used memory to maximum of 4GB');
+				$memory = 4 * 1024 * 1024 * 1024;
+			}
 		}
 
+		// Apply admin setting limit.
+		$memorySetting = $this->config->getAppValue('facerecognition', 'memory-limits', '-1');
+		if ($memorySetting > 0) {
+			if ($memorySetting > $memory) {
+				$error_message =
+					"\n" .
+					"Seems that you configured" . intval($memorySetting / (1024 * 1024)) . " MB of memory, however, this exede"
```

Comment:
exede <=> exceeds

Reply:
> if you don't use this value if $status is STATE_ERROR, why do you set it here then?

I could take it into account in the fronted as the suggested value.. But It just does not bother, the important thing is to indicate that the configuration is invalid.